Expose WebCodecs VideoDecoder Skeleton

youennf opened the pull request:
#### 4e6b9305bf345d0ea284e44971f3e4f88018d56d
<pre>
Expose WebCodecs VideoDecoder Skeleton
<a href="https://bugs.webkit.org/show_bug.cgi?id=245730">https://bugs.webkit.org/show_bug.cgi?id=245730</a>
rdar://problem/100457656

Reviewed by Eric Carlson.

Add infrastructure and VideoDecoder skeleton implementation

* LayoutTests/imported/w3c/web-platform-tests/webcodecs/video-decoder.crossOriginIsolated.https.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/video-decoder.crossOriginIsolated.https.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/video-decoder.https.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/video-decoder.https.any.worker-expected.txt:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/webcodecs/HardwareAcceleration.h: Added.
* Source/WebCore/Modules/webcodecs/HardwareAcceleration.idl: Added.
* Source/WebCore/Modules/webcodecs/WebCodecsCodecState.h: Added.
* Source/WebCore/Modules/webcodecs/WebCodecsCodecState.idl: Added.
* Source/WebCore/Modules/webcodecs/WebCodecsErrorCallback.h: Added.
* Source/WebCore/Modules/webcodecs/WebCodecsErrorCallback.idl: Added.
* Source/WebCore/Modules/webcodecs/WebCodecsVideoDecoder.cpp: Added.
(WebCore::WebCodecsVideoDecoder::create):
(WebCore::WebCodecsVideoDecoder::WebCodecsVideoDecoder):
(WebCore::WebCodecsVideoDecoder::~WebCodecsVideoDecoder):
(WebCore::WebCodecsVideoDecoder::isConfigSupported):
(WebCore::WebCodecsVideoDecoder::close):
(WebCore::WebCodecsVideoDecoder::flush):
(WebCore::WebCodecsVideoDecoder::reset):
(WebCore::WebCodecsVideoDecoder::decode):
(WebCore::WebCore::WebCodecsVideoDecoder::suspend):
(WebCore::WebCodecsVideoDecoder::configure):
(WebCore::WebCodecsVideoDecoder::stop):
(WebCore::WebCodecsVideoDecoder::activeDOMObjectName const):
(WebCore::WebCodecsVideoDecoder::virtualHasPendingActivity const):
* Source/WebCore/Modules/webcodecs/WebCodecsVideoDecoder.h: Added.
(WebCore::WebCodecsVideoDecoder::state const):
(WebCore::WebCodecsVideoDecoder::decodeQueueSize const):
* Source/WebCore/Modules/webcodecs/WebCodecsVideoDecoder.idl: Added.
* Source/WebCore/Modules/webcodecs/WebCodecsVideoDecoderConfig.h: Added.
* Source/WebCore/Modules/webcodecs/WebCodecsVideoDecoderConfig.idl: Added.
* Source/WebCore/Modules/webcodecs/WebCodecsVideoDecoderSupport.h: Added.
* Source/WebCore/Modules/webcodecs/WebCodecsVideoDecoderSupport.idl: Added.
* Source/WebCore/Modules/webcodecs/WebCodecsVideoFrameOutputCallback.h: Added.
* Source/WebCore/Modules/webcodecs/WebCodecsVideoFrameOutputCallback.idl: Added.
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:
* Source/WebCore/dom/EventNames.h:
* Source/WebCore/dom/EventTargetFactory.in:

Canonical link: <a href="https://commits.webkit.org/255118@main">https://commits.webkit.org/255118@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9732f9236430bd7fda0831c173c00f37af389bef

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/91426 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/483 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/22074 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/101152 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/161140 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/95431 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/498 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/29351 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/83773 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/97521 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/97084 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/361 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/61/builds/78145 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/27325 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/82285 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/81947 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/70356 "Found 1 new API test failure: /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/text/state-changed (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/35561 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/15974 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/33346 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/17064 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3561 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/37146 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/61/builds/78145 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/39066 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/36178 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

eric-carlson commented on Source/WebCore/Modules/webcodecs/WebCodecsErrorCallback.h:
```diff
@@ -0,0 +1,52 @@
+
+    // Helper to post callback task.
+    void scheduleCallback(ScriptExecutionContext&, Ref<DOMException>&&);
+};
```
Did you forget to include WebCodecsErrorCallback.cpp?

youennf replied:
scheduleCallback is not needed, I removed it from the latest patch.

eric-carlson reviewed (approved):
r=me

youennf committed:
Expose WebCodecs VideoDecoder Skeleton
https://bugs.webkit.org/show_bug.cgi?id=245730
rdar://problem/100457656

Reviewed by Eric Carlson.

Add infrastructure and VideoDecoder skeleton implementation

* LayoutTests/imported/w3c/web-platform-tests/webcodecs/video-decoder.crossOriginIsolated.https.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/video-decoder.crossOriginIsolated.https.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/video-decoder.https.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/video-decoder.https.any.worker-expected.txt:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/webcodecs/HardwareAcceleration.h: Added.
* Source/WebCore/Modules/webcodecs/HardwareAcceleration.idl: Added.
* Source/WebCore/Modules/webcodecs/WebCodecsCodecState.h: Added.
* Source/WebCore/Modules/webcodecs/WebCodecsCodecState.idl: Added.
* Source/WebCore/Modules/webcodecs/WebCodecsErrorCallback.h: Added.
* Source/WebCore/Modules/webcodecs/WebCodecsErrorCallback.idl: Added.
* Source/WebCore/Modules/webcodecs/WebCodecsVideoDecoder.cpp: Added.
(WebCore::WebCodecsVideoDecoder::create):
(WebCore::WebCodecsVideoDecoder::WebCodecsVideoDecoder):
(WebCore::WebCodecsVideoDecoder::~WebCodecsVideoDecoder):
(WebCore::WebCodecsVideoDecoder::isConfigSupported):
(WebCore::WebCodecsVideoDecoder::close):
(WebCore::WebCodecsVideoDecoder::flush):
(WebCore::WebCodecsVideoDecoder::reset):
(WebCore::WebCodecsVideoDecoder::decode):
(WebCore::WebCore::WebCodecsVideoDecoder::suspend):
(WebCore::WebCodecsVideoDecoder::configure):
(WebCore::WebCodecsVideoDecoder::stop):
(WebCore::WebCodecsVideoDecoder::activeDOMObjectName const):
(WebCore::WebCodecsVideoDecoder::virtualHasPendingActivity const):
* Source/WebCore/Modules/webcodecs/WebCodecsVideoDecoder.h: Added.
(WebCore::WebCodecsVideoDecoder::state const):
(WebCore::WebCodecsVideoDecoder::decodeQueueSize const):
* Source/WebCore/Modules/webcodecs/WebCodecsVideoDecoder.idl: Added.
* Source/WebCore/Modules/webcodecs/WebCodecsVideoDecoderConfig.h: Added.
* Source/WebCore/Modules/webcodecs/WebCodecsVideoDecoderConfig.idl: Added.
* Source/WebCore/Modules/webcodecs/WebCodecsVideoDecoderSupport.h: Added.
* Source/WebCore/Modules/webcodecs/WebCodecsVideoDecoderSupport.idl: Added.
* Source/WebCore/Modules/webcodecs/WebCodecsVideoFrameOutputCallback.h: Added.
* Source/WebCore/Modules/webcodecs/WebCodecsVideoFrameOutputCallback.idl: Added.
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:
* Source/WebCore/dom/EventNames.h:
* Source/WebCore/dom/EventTargetFactory.in:

Canonical link: https://commits.webkit.org/255118@main
